### PR TITLE
Convert from FunScript to Fable

### PR DESCRIPTION
**Pull request by alfonsogarciacaro**
Here comes my first attempt to convert the awesome Ionide plugins to Fable!

I've used your `vscode-fable-demo` as a basis. Basically the changes are:
- For simplicity I've converted the project to a single .fsx file and removed the now unneeded files.
- Also for simplicity, I just copied the `Helper` modules into the script. I'll update the `ionide-vscode-helpers` later.
- I had to update the Fable Node definitions to fix some mistakes.
- I'm using the full name to reference a couple of modules that were now conflicting (like `Promise`).
- I've added another helper function (within the `activate` method) to register commands.

On a quick test, it seems to work :+1:  Can you please have a look and tell me what you think? If you want me to make some changes, just tell me.

Cheers!

### CODE REVIEW

**Comment by Krzysztof-Cieslak** [comment]
That looks really good!

Only thing that bothers me is that you completely removed FAKE build scripts - I think I would want to leave parts responsible for versioning, packaging and publishing as they were. Probably whole script should be almost the same, only difference would be calling `npm run build` instead of old generation.

**Comment by alfonsogarciacaro** [comment]
You're right. I was just trying to find the minimal structure to build the extension with Fable but it seems I went too far ;) I'll put the FAKE script back :+1:

**Comment by alfonsogarciacaro** [comment]
Done! Please check if the changes I did in the build.fsx script make sense. Cheers!

File: build.fsx [hunk 1]
```diff
@@ -73,35 +66,20 @@ let vsceTool =
 
 Target "Clean" (fun _ ->
     CleanDir "./temp"
-    CopyFiles "release" ["README.md"; "LICENSE.md"; "RELEASE_NOTES.md"]
```
Comment (forki) [inline build.fsx:76]: what about these files?

Comment (alfonsogarciacaro) [inline build.fsx:76]: They were just copies of the files in the root dir, but now the `release` dir is the root itself so the copies shouldn't be needed anymore. Of course, if @Krzysztof-Cieslak prefers to still have a `release` dir we can bring them back.

Comment (Krzysztof-Cieslak) [inline build.fsx:76]: I think I would prefer leaving release `dir` as it is, at least for this first PR. When we get everything working (whole release process) we can think about making repository setup bit better. But let's do small steps now.

Comment (alfonsogarciacaro) [inline build.fsx:76]: No problem! I'll put the `release` folder back :+1:

Comment (alfonsogarciacaro) [inline build.fsx:76]: Sorry for the delay, `release` folder is back again! :)

**Comment by Krzysztof-Cieslak** [comment]
Before we go with it, I would like to convert Helpers project to Fable so we don't have Helpers module here. I guess we can just create Fable branch on Helpers repo and put this code there for start.

**Comment by alfonsogarciacaro** [comment]
Perfect! Please create the branch so I can send the PR there 👍 Also, how do you want to interact with the Helpers from the other packages? Just compiling the code into the final JS or do you want to make a npm package for the Helpers? This could prevent some duplicity of code (if VS Code extensions can share dependencies) but for that we will probably have to use F# projects instead of scripts.

**Comment by cloudRoutine** [comment]
@alfonsogarciacaro  can you pr onto https://github.com/ionide/ionide-vscode-paket/tree/fable instead?

**Comment by Krzysztof-Cieslak** [comment]
And Helpers part to https://github.com/ionide/ionide-vscode-helpers/tree/fable. :)

**Comment by cloudRoutine** [comment]
I merged the PR onto this branch https://github.com/ionide/ionide-vscode-paket/tree/fable

**Pull request closed without merging**